### ISSUE TITLE
Creating a network namespace within a systemd-nspawn container

### ISSUE THREAD

**Issue by floppym** [systemd/systemd#13308]
On Gentoo Linux, our package build tool (Portage) has a feature that creates an isolated network namespace while building the software. This prevents build systems from fetching random files from the internet during the build process.

When I run Portage in a systemd-nspawn container with private-network disabled, Portage emits the following errors:
```
Unable to enable loopback interface: Operation not permitted
RTNETLINK answers: Operation not permitted
RTNETLINK answers: Operation not permitted
```
This comes from this code in Portage.
https://github.com/gentoo/portage/blob/portage-2.3.71/lib/portage/process.py#L487

Basically, we get `EPERM` when we try to bring up the `lo` interface in the network namespace. I believe this happens because systemd-nspawn removes the `CAP_NET_ADMIN` capability when it spawns the container.

Is there any (secure) way to both allow a container to inherit the network configuration of the host system, and also allow the container to create network namespaces and configure interfaces within those namespaces?

**Comment by poettering** [comment]
i figure you should be able to turn on user namespacing. But that comes with other caveats...

you could also use `--capability=CAP_NET_ADMIN` I figure, but that's not secure

**Comment by lkraav** [comment]
I'm running into the same thing. Trying to get Bash built-in `${HOSTNAME}` variable to display in a `/etc/portage/bashrc` hook, but while it works, has a side effect of also echoing "Unable to configure loopback interface: Operation not permitted".

I wonder if there's a minimal side-effect way of at least killing the notice off?

**Comment by migrgh** [comment]
Same issue here. My solution would be to just run emerge-webrsync and portage with --fetchonly to obtain the source files with network and then for emerge (build packages) with nspawn --private-network .
